samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -202,7 +202,7 @@ static int subscribe(void)
 static int publish_get_payload(struct mqtt_client *c, size_t length)
 {
 	if (length > sizeof(payload_buf)) {
-		return -EMSGSIZE;
+		LOG_ERR("Payload is larger than payload buffer. Truncating received message.");
```

Comment:
I don't think we can simply ignore this fact, `mqtt_readall_publish_payload()` will write out of the buffer bounds in this case.

Reply:
True, thanks for noticing! I will rewrite this to actually truncate the response without overflowing the buffer. Let's see how it looks and then decide if we want to include it or let the fix in #6371 be enough. #6371 will fix what we observe now, with a retained message on the subscribe topic, having a payload of 350 bytes. However, larger payloads than the 400 bytes set in #6371 can be published as well.